thoth-api/src/model/file/mod.rs

```diff
@@ -0,0 +1,240 @@
+    pub work_id: Option<Uuid>,
+    pub publication_id: Option<Uuid>,
+    pub declared_mime_type: String,
+    pub declared_extension: String,
+    pub declared_sha256: String,
+}
+
+#[cfg_attr(
+    feature = "backend",
+    derive(Insertable),
+    diesel(table_name = file)
+)]
+pub struct NewFile {
+    pub file_type: FileType,
+    pub work_id: Option<Uuid>,
+    pub publication_id: Option<Uuid>,
+    pub object_key: String,
+    pub cdn_url: String,
+    pub mime_type: String,
+    pub bytes: i64,
+    pub sha256: String,
+}
+
+#[cfg(feature = "backend")]
+#[derive(juniper::GraphQLInputObject)]
+#[graphql(description = "Input for starting a publication file upload (PDF, EPUB, XML, etc.).")]
+pub struct NewPublicationFileUpload {
+    #[graphql(description = "Thoth ID of the publication linked to this file.")]
+    pub publication_id: Uuid,
+    #[graphql(
+        description = "MIME type declared by the client (used for validation and in the presigned URL)."
+    )]
+    pub declared_mime_type: String,
+    #[graphql(
+        description = "File extension to use in the final canonical key, e.g. 'pdf', 'epub', 'xml'."
+    )]
+    pub declared_extension: String,
+    #[graphql(description = "SHA-256 checksum of the file, hex-encoded.")]
+    pub declared_sha256: String,
+}
+
+#[cfg(feature = "backend")]
+#[derive(juniper::GraphQLInputObject)]
+#[graphql(description = "Input for starting a front cover upload for a work.")]
+pub struct NewFrontcoverFileUpload {
+    #[graphql(description = "Thoth ID of the work this front cover belongs to.")]
+    pub work_id: Uuid,
+    #[graphql(description = "MIME type declared by the client (e.g. 'image/jpeg').")]
+    pub declared_mime_type: String,
+    #[graphql(
+        description = "File extension to use in the final canonical key, e.g. 'jpg', 'png', 'webp'."
+    )]
+    pub declared_extension: String,
+    #[graphql(description = "SHA-256 checksum of the file, hex-encoded.")]
+    pub declared_sha256: String,
+}
+
+#[cfg(feature = "backend")]
+#[derive(juniper::GraphQLInputObject)]
+#[graphql(
+    description = "Input for completing a file upload and promoting it to its final DOI-based location."
+)]
+pub struct CompleteFileUpload {
+    #[graphql(description = "ID of the upload session to complete.")]
+    pub file_upload_id: Uuid,
+}
+
+#[cfg(feature = "backend")]
+#[derive(juniper::GraphQLObject)]
+#[graphql(
+    description = "Response from initiating a file upload, containing the upload URL and expiration time."
+)]
+pub struct FileUploadResponse {
+    #[graphql(description = "ID of the upload session.")]
+    pub file_upload_id: Uuid,
+    #[graphql(description = "Presigned S3 PUT URL for uploading the file.")]
+    pub upload_url: String,
+    #[graphql(description = "Time when the upload URL expires.")]
+    pub expires_at: Timestamp,
+}
+
+#[cfg(feature = "backend")]
+/// Parse a DOI into prefix and suffix
+pub fn parse_doi(doi: &Doi) -> ThothResult<(String, String)> {
```

Comment:
The `Doi` struct already performs validation, implementing `FromStr` (check `thoth-api/src/model/mod.rs`), and can output the string without the domain using `to_string()` (`impl Display`).

Overall, it'd make more sense to add a couple of methods to `impl Doi` in the model: `fn prefix(&self) -> String` and `fn suffix(&self) -> String`